projects/vdk-core/src/vdk/internal/builtin_plugins/job_properties/base_properties_impl.py

```diff
@@ -1,13 +1,14 @@
 # Copyright 2021-2023 VMware, Inc.
 # SPDX-License-Identifier: Apache-2.0
-from typing import List
-from typing import Type
 
 
-def check_valid_property(k: str, v: str, supported_types: List[Type] = []) -> None:
+def check_valid_property(k: str, v: str, supported_types=None) -> None:
```

Comment:
why did you removed the type info? 
Also why do you set the default to None. But then the first thing you do if it is None is set it to empty list.

Reply:
Codacy check suggestion

Reply:
To be more precise , the codacy warning is because Python evaluates the default argument once and then uses the same "instance" of that argument every time the function is called (if the argument is not provided). 

Basically acts as a global variable. 

See: https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/dangerous-default-value.html